cmd/lotus-miner/actor.go

```diff
@@ -1,1452 +1,45 @@
-
-		if (cctx.IsSet("existing-beneficiary") && cctx.IsSet("new-beneficiary")) || (!cctx.IsSet("existing-beneficiary") && !cctx.IsSet("new-beneficiary")) {
-			return lcli.ShowHelp(cctx, fmt.Errorf("must pass exactly one of --existing-beneficiary or --new-beneficiary"))
-		}
-
-		var fromAddr address.Address
-		if cctx.IsSet("existing-beneficiary") {
-			if mi.PendingBeneficiaryTerm.ApprovedByBeneficiary {
-				return fmt.Errorf("beneficiary change already approved by current beneficiary")
-			}
-			fromAddr = mi.Beneficiary
-		} else {
-			if mi.PendingBeneficiaryTerm.ApprovedByNominee {
-				return fmt.Errorf("beneficiary change already approved by new beneficiary")
-			}
-			fromAddr = mi.PendingBeneficiaryTerm.NewBeneficiary
-		}
-
-		fmt.Println("Confirming Pending Beneficiary Term of:")
-		fmt.Println("Beneficiary: ", mi.PendingBeneficiaryTerm.NewBeneficiary)
-		fmt.Println("Quota:", mi.PendingBeneficiaryTerm.NewQuota)
-		fmt.Println("Expiration Epoch:", mi.PendingBeneficiaryTerm.NewExpiration)
-
-		if !cctx.Bool("really-do-it") {
-			fmt.Println("Pass --really-do-it to actually execute this action. Review what you're about to approve CAREFULLY please")
-			return nil
-		}
-
-		params := &miner.ChangeBeneficiaryParams{
-			NewBeneficiary: mi.PendingBeneficiaryTerm.NewBeneficiary,
-			NewQuota:       mi.PendingBeneficiaryTerm.NewQuota,
-			NewExpiration:  mi.PendingBeneficiaryTerm.NewExpiration,
-		}
-
-		sp, err := actors.SerializeParams(params)
-		if err != nil {
-			return xerrors.Errorf("serializing params: %w", err)
-		}
-
-		smsg, err := api.MpoolPushMessage(ctx, &types.Message{
-			From:   fromAddr,
-			To:     maddr,
-			Method: builtin.MethodsMiner.ChangeBeneficiary,
-			Value:  big.Zero(),
-			Params: sp,
-		}, nil)
-		if err != nil {
-			return xerrors.Errorf("mpool push: %w", err)
-		}
-
-		fmt.Println("Confirm Message CID:", smsg.Cid())
-
-		// wait for it to get mined into a block
-		wait, err := api.StateWaitMsg(ctx, smsg.Cid(), build.MessageConfidence)
-		if err != nil {
-			return xerrors.Errorf("waiting for message to be included in block: %w", err)
-		}
-
-		// check it executed successfully
-		if wait.Receipt.ExitCode.IsError() {
-			return fmt.Errorf("confirm beneficiary change failed with code %d", wait.Receipt.ExitCode)
-		}
-
-		updatedMinerInfo, err := api.StateMinerInfo(ctx, maddr, types.EmptyTSK)
-		if err != nil {
-			return err
-		}
-
-		if updatedMinerInfo.PendingBeneficiaryTerm == nil && updatedMinerInfo.Beneficiary == mi.PendingBeneficiaryTerm.NewBeneficiary {
-			fmt.Println("Beneficiary address successfully changed")
-		} else {
-			fmt.Println("Beneficiary address change awaiting additional confirmations")
-		}
-
-		return nil
+		spcli.ActorSetAddrsCmd(LMActorGetter),
+		spcli.ActorWithdrawCmd(LMActorGetter),
+		spcli.ActorRepayDebtCmd(LMActorGetter),
+		spcli.ActorSetPeeridCmd(LMActorGetter),
+		spcli.ActorSetOwnerCmd(LMConfigOrActorGetter),
+		spcli.ActorControlCmd(LMConfigOrActorGetter, actorControlList),
```

Comment:
I think we need to wrap this with flag to supply the local miner config.toml path

Reply:
I don't see anywhere that the config.toml is used in these 2 sub-commands, so I think it's ok.